### PR TITLE
Feature/add same site

### PR DESCRIPTION
**Pull request by vvval**
- add `sameSite` attribute to cookie queue
- polish code

### CODE REVIEW

**Comment by vvval** [comment]
This PR is the 1st step, need to improve spiral/csrf, spiral/auth-http, spiral/framework packages after that

**Comment by SerafimArts** [comment]
@vvval why not create a separate enum-like interface for the samesite types? Like:

```php
interface SameSite
{
    public const SAME_SITE_STRICT = 'Strict';
    public const SAME_SITE_LAX = 'Lax';
    public const SAME_SITE_NONE = 'None';
}
```
 
And inside the Cookie object, use private constants that would take values from these constants. In addition, the object itself can implement this interface in order to make it easier to use them (this is why I added prefixes `SAME_SITE_*` to the interface).

In addition, the cookie object can be divided according to the principle of ISP, for example:

```php
interface SameSiteInterface
{
    public const SAME_SITE_STRICT = 'Strict';
    public const SAME_SITE_LAX = 'Lax';
    public const SAME_SITE_NONE = 'None';

    public function setSameSite(?string $type): void;
}

interface CookieInterface extends SameSiteInterface, HttpOnlyeInterface, etc.... {}

final class Cookie implements CookieInterface
{
}
```

In addition, such external functionality can be divided into traits, like `SameSiteTrait` implementation for `SameSiteInterface`. A similar approach is used, for example, in [Zend Diactoros](https://github.com/laminas/laminas-diactoros/tree/master/src)

**Comment by vvval** [comment]
@SerafimArts Not sure that the cookie should be split up into separate interfaces, all these mentioned flags (`secure`, `httpOnly`, `sameSite`) do not exist outside of this class and are used only internally just to pack the properties into the header line. I even think that we need to cut off some public getters, but its a risky situation with a 100% BC-break. We can add this to a new milestone in the next version.

But I agree that we can extract `sameSite` not as a simple enum but as a separate object with its own state and transforming logic

**Comment by vvval** [comment]
@SerafimArts please revise the new object introduction

File: src/Cookie/SameSite.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spiral\Cookies\Cookie;
+
+class SameSite
```
Comment (SerafimArts) [inline src/Cookie/SameSite.php:7]: What do you think about add `isSecure(): bool` method?

**Review by SerafimArts — APPROVED** [review]
My bad. Found it: https://github.com/spiral/cookies/blob/c11f6112cdf78f15f348c7e8027b238cb372d69e/src/Cookie.php#L204

File: src/Cookie/SameSite.php (hunk 1 above)
Comment (vvval) [reply src/Cookie/SameSite.php:7]: It's not used, I need it just to define the correct `sameSite` value because `sameSite` depends on `isSecure`

**Pull request merged**